packages/context/test/acceptance/creating-and-resolving-bindings.ts

```diff
@@ -34,6 +34,26 @@ describe('Context bindings - Creating and resolving bindings', () => {
         }
       });
     });
+
+    context('with type information', () => {
+      it('infers correct type when getting the value', async () => {
+        const key = new BindingKey<string>('foo');
+        ctx.bind(key).to('value');
+        const value = await ctx.get(key);
+        // The following line is accessing a String property as a way
+        // of verifying the value type at compile time
+        expect(value.length).to.equal(5);
```

Comment:
Isn't it also fair to just assert `expect(value).to.be.a.String()`?

Reply:
Ignore this comment and the next if the approach above has no way of verifying the type at compile time

Reply:
Unfortunately `expect(value)` accepts any type at compile time, it cannot be used to make any compile-time assertions.